python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll

```diff
@@ -388,23 +388,27 @@ module LocalFlow {
 module VariableCapture {
   private import codeql.dataflow.VariableCapture as Shared
 
-  class ExprCfgNode extends ControlFlowNode {
-    ExprCfgNode() { isExpressionNode(this) }
-  }
-
-  private predicate closureFlowStep(ExprCfgNode nodeFrom, ExprCfgNode nodeTo) {
-    // TODO: Other languages have an extra case here looking like
-    //   simpleAstFlowStep(nodeFrom, nodeTo)
-    // we should investigate the potential benefit of adding that.
-    exists(SsaVariable def |
-      def.getAUse() = nodeTo and
-      def.getAnUltimateDefinition().getDefinition().(DefinitionNode).getValue() = nodeFrom
-    )
-  }
-
   private module CaptureInput implements Shared::InputSig<Location> {
     private import python as PY
 
+    additional class ExprCfgNode extends ControlFlowNode {
+      ExprCfgNode() { isExpressionNode(this) }
+    }
+
+    private predicate closureFlowStep(ExprCfgNode nodeFrom, ExprCfgNode nodeTo) {
+      // TODO: Other languages have an extra case here looking like
+      //   simpleAstFlowStep(nodeFrom, nodeTo)
+      // we should investigate the potential benefit of adding that.
+      exists(SsaVariable def |
+        def.getAUse() = nodeTo and
+        def.getAnUltimateDefinition().getDefinition().(DefinitionNode).getValue() = nodeFrom
+      )
+    }
```

Comment:
can we move this next to where it's used? :pray:

Reply:
Done